streams/src/test/java/org/apache/kafka/streams/integration/QueryableStateIntegrationTest.java

```diff
@@ -791,12 +788,16 @@ public boolean conditionMet() {
 
         final ReadOnlyKeyValueStore<String, String> store2 = kafkaStreams.store(storeName, QueryableStoreTypes.<String, String>keyValueStore());
 
-        TestUtils.waitForCondition(new TestCondition() {
-            @Override
-            public boolean conditionMet() {
-                return "125".equals(store2.get("a")) && "34".equals(store2.get("b"));
-            }
-        }, maxWaitMs, "wait for agg to be '123'");
+        try {
+            TestUtils.waitForCondition(new TestCondition() {
+                @Override
+                public boolean conditionMet() {
+                    return store2.get("a").endsWith("125") && store2.get("b").endsWith("34");
+                }
+            }, maxWaitMs, "wait for agg to be <a,125> and <b,34>");
```

Comment:
That seems right.. Hey I actually am not thinking of a even smaller fix: since this `shouldAllowToQueryAfterThreadDied` is only to validate that after the thread dies, we do not actually need to rely on data at all. I.e. the state store can successfully be migrated and be queried. What we can do is that:

```
                    if (value1.equals("dummy") && beforeFailure.compareAndSet(true, false)) {
                            throw new RuntimeException("Injected test exception");
                    } else if (failed.get()) {
                            return "resumed!";
                    } else {
                            return "normal";
                    }
```

And then we can send just a single dummy record from the async producer, and the intermediate and the final condition just check that `store2.get("a").equals("normal") && store2.get("b").equals("normal")` and `store2.get("a").equals("resumed!") && store2.get("b").equals("resumed!")`. WDYT?